### PR TITLE
Check digest header

### PR DESCRIPTION
**Pull request by yorikim**

### CODE REVIEW

File: lib/rack/http/signatures/digest.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    attr_reader :body
+
+    def initialize(algorithm, body)
+      @algorithm = algorithm
+      @body = body
+    end
+
+    def valid_algorithm?
+      VALID_ALGORITHMS.include?(@algorithm)
```
Comment (rjnienaber) [inline lib/rack/http/signatures/digest.rb]: Should we ignore case when comparing?

File: spec/rack/http/signatures/verify_signature_spec.rb [hunk 2]
```diff
@@ -12,52 +12,128 @@
...
-     :'hmac-sha256' => 'NzU4ZDAyMGRmNzQxZmQ3NDQ0YWY0Mzk5Y2YxMjUzYzA1NGI2MWQ2OTc5NjhlYjM3NTg2Y2I1MmFiMDlkN2NkNA=='
-    }.each do |algorithm, signature|
-      context "using #{algorithm}" do
-        context 'with headers field' do
-          it 'returns success code when keyId is valid' do
-            response = request.get('/', http_headers.merge('HTTP_AUTHORIZATION' => "Signature keyId=\"Test\",algorithm=\"#{algorithm}\",headers=\"date\",signature=\"#{signature}\""))
+    context 'check digest header' do
+      it 'returns 400 when signature is invalid' do
+        http_headers['HTTP_DIGEST'] = 'SHA-256=INVALID='
+        response = request.post(request_path, http_headers.merge('HTTP_AUTHORIZATION' => "Signature keyId=\"Test\",algorithm=\"rsa-sha256\",signature=\"jKyvPcxB4JbmYY4mByyBY7cZfNl4OW9HpFQlG7N4YcJPteKTu4MWCLyk+gIr0wDgqtLWf9NLpMAMimdfsH7FSWGfbMFSrsVTHNTk0rK3usrfFnti1dxsM4jl0kYJCKTGI/UWkqiaxwNiKqGcdlEDrTcUhhsFsOIo8VhddmZTZ8w=\""))
```
Comment (rjnienaber) [inline spec/rack/http/signatures/verify_signature_spec.rb]: If the ```HTTP_AUTHORIZATION``` header is identical for the other tests, can we specify it once during setup as ```let``` ?

File: lib/rack/http/signatures/digest.rb (hunk 1 above)
Comment (yorikim) [inline lib/rack/http/signatures/digest.rb]: Yes, we should.
https://tools.ietf.org/html/rfc3230#section-4.1.1

File: spec/rack/http/signatures/verify_signature_spec.rb (hunk 2 above)
Comment (yorikim) [inline spec/rack/http/signatures/verify_signature_spec.rb]: It's different for each case.

File: lib/rack/http/signatures/digest.rb (hunk 1 above)
Comment (rjnienaber) [inline lib/rack/http/signatures/digest.rb]: 👍

Comment (yorikim) [inline lib/rack/http/signatures/digest.rb]: fixed

File: spec/rack/http/signatures/verify_signature_spec.rb (hunk 2 above)
Comment (yorikim) [inline spec/rack/http/signatures/verify_signature_spec.rb]: removed duplicates in this pr. https://github.com/CurrencyCloud/rack-http-signatures/pull/16

**Pull request merged**